apps/aemon/src/aemon_mon_ttl.erl

```diff
@@ -0,0 +1,79 @@
+-module(aemon_mon_ttl).
+-behaviour(gen_server).
+
+-export([publisher_tx/2]).
+-export([on_chain_tx/1, on_chain_height/1]).
+
+-export([start_link/0]).
+-export([ init/1
+        , handle_call/3
+        , handle_cast/2
+        , handle_info/2
+        , terminate/2
+        , code_change/3
+        ]).
+
+-record(state, {  publisher_height = 0
+                , publisher_txs = #{}
+               }).
+
+publisher_tx(Height, Tx) ->
+    gen_server:cast(?MODULE, {tx, Height, Tx}).
+
+on_chain_tx(TxHash) ->
+    gen_server:cast(?MODULE, {on_chain_tx, TxHash}).
+
+on_chain_height(Height) ->
+    gen_server:cast(?MODULE, {on_chain_height, Height}).
+
+start_link() ->
+    gen_server:start_link({local, ?MODULE}, ?MODULE, [], []).
+
+%% gen_server callback
+
+
+init(_) ->
+    aemon_metrics:create(ttl),
+    {ok, #state{}}.
+
+terminate(_Reason, _St) ->
+    ok.
+
+code_change(_FromVsn, St, _Extra) ->
+    {ok, St}.
+
+handle_call(_Req, _From, St) ->
+    {reply, {error, unknown_request}, St}.
+
+handle_cast({tx, Height, SignTx}, St0) ->
+    St = txs_size_metric(Height, St0),
+    Hash = aetx_sign:hash(SignTx),
+    TTL = aetx:ttl(aetx_sign:tx(SignTx)),
+    NewTxs = maps:put(Hash, TTL, St#state.publisher_txs),
+    {noreply, St#state{publisher_txs = NewTxs}};
+
+handle_cast({on_chain_tx, TxHash}, St = #state{publisher_txs = Txs}) ->
+    NewTxs = maps:remove(TxHash, Txs),
+    {noreply, St#state{publisher_txs = NewTxs}};
+
+handle_cast({on_chain_height, Height}, St = #state{publisher_txs = Txs}) ->
+    NewTxs = maps:filter(fun(_, AccHeight) -> AccHeight >= Height end, Txs),
```

Comment:
```
2> max_ttl > 1234567890.
true
```
Can `AccHeight` be `max_ttl`?
* If yes, does it mean that if txs never included on chain then memory of this process grows indefinitely?
* If no - e.g. because of how tx is posted (did not check that point) - could you please make this clause stricter e.g. 
```suggestion
    NewTxs = maps:filter(fun(_, AccHeight) when is_integer(AccHeight), AccHeight > 0 -> AccHeight >= Height end, Txs),
```

Reply:
based on `aemon_publisher` `AccHeight` cant be equal to `max_ttl`

Reply:
"cant" - thanks. (Initially I had not seen the "t" :) .) I see the removal of `max_ttl` from spec - thanks.